Python/pystate.c

```diff
@@ -1034,6 +1034,11 @@ PyThreadState_Clear(PyThreadState *tstate)
     Py_CLEAR(tstate->async_gen_finalizer);
 
     Py_CLEAR(tstate->context);
+    
+    if (tstate->ref != NULL)
```

Comment:
I'm not sure how PyThreadState interacts with the garbage collector. How will the GC know that the thread state ref object is alive?

Reply:
The ref object has a life on its own. The PyThreadState creates the ref object. Once the thread state is deallocated, it sets the actual ref inside the ref object to NULL and decrements its refcount. The ref object then lives on until there are references to it. Otherwise, it is GC'd at the next collection cycle.